### PR TITLE
Test for SVHN

### PR DESCRIPTION
**Pull request by pmeier**
This moves the cast from `numpy.ndarray` to `PIL.Image` from `__getitem__()` to `__init__()` for the `CIFAR10`, `CIFAR10`, and `SVHN` dataset. Additionally, this adds a dataset test for the `SVHN` dataset.

### CODE REVIEW

File: torchvision/datasets/cifar.py [hunk 1]
```diff
@@ -94,6 +94,10 @@ def __init__(self, root, train=True,
         self.data = np.vstack(self.data).reshape(-1, 3, 32, 32)
         self.data = self.data.transpose((0, 2, 3, 1))  # convert to HWC
 
+        # doing this so that it is consistent with all other datasets
+        # to return a PIL Image
+        self.data = [Image.fromarray(image) for image in self.data]
```
Comment (fmassa) [inline torchvision/datasets/cifar.py]: This is a breaking change if users accessed the internal field `.data` before.

I wonder if we should instead add a deprecation warning in this case?

**Comment by pmeier** [comment]
> So it would be good to know how much slowdown this represents.

It takes ~80µs per image (on my machine) for the conversion, thus making the performance advantage negligible. I would argue that the performance was never the decisive factor for this PR. I simply found it odd that we keep the data as `numpy.ndarray` while we internally use `PIL` as image backend. But then I'm from a _prototypy_ world and keep forgetting that BC could be more important.

If you think BC is more important here, I'm happy to roll back the changes and only keep the test for `SVHN`.

File: torchvision/datasets/cifar.py (hunk 1 above)
Comment (pmeier) [reply torchvision/datasets/cifar.py]: Should `data` be an internal field? I would argue in favour of this, but I wasn't sure if that is the case fore everyone.

Comment (fmassa) [reply torchvision/datasets/cifar.py]: Even if we do not encourage users to query it, they still do it, see for example https://github.com/pytorch/vision/pull/578#issuecomment-460615581

**Comment by fmassa** [comment]
@pmeier I think it's important to keep BC until we have a clear story about what are the fields of datasets that users can use, and those that they shouldn't rely on. Your issue on unifying datasets is a great start, and I'll make some comments today on it.

If you could revert the BC changes and keep only the test for `SVHN` for now, this would be ready to merge

File: torchvision/datasets/cifar.py (hunk 1 above)
Comment (pmeier) [reply torchvision/datasets/cifar.py]: We should take this up in #1080.
